lib/charms/prometheus_k8s/v0/prometheus_remote_write.py

```diff
@@ -595,6 +595,8 @@ def __init__(
         charm: CharmBase,
         relation_name: str = DEFAULT_CONSUMER_NAME,
         alert_rules_path: str = DEFAULT_ALERT_RULES_RELATIVE_PATH,
+        *,
+        extra_alerts_callable: Optional[Callable] = None,
```

Comment:
We should probably mention motivation+usage of this arg somewhere in the docstring.

I'd also add typing. I think it's
```python
extra_alerts_callable: Optional[Callable[[], dict]] = None,
```
?

Reply:
I think we need to specify `Any` as well. Done

Reply:
fixed